stubs/setuptools/pkg_resources/__init__.pyi

```diff
@@ -243,19 +253,49 @@ def register_loader_type(loader_type: type, provider_factory: Callable[[types.Mo
 def register_namespace_handler(importer_type: type, namespace_handler: _NSHandlerType) -> None: ...
 
 class IResourceProvider(IMetadataProvider): ...
-class NullProvider: ...
-class EggProvider(NullProvider): ...
+
+class NullProvider:
+    egg_name: str | None
+    egg_info: str | None
+    loader: types._LoaderProtocol | None
+    module_path: str | None
```

Comment:
```suggestion
    module_path: str
```
The initializer unconditionally sets it to a string.

Reply:
EmptyProvider sets it explicitely to None

Should I keep `module_path: str | None` for sublcassing. Or add `# type: error[assignment]` to `EmptyProvider.module_path` ?